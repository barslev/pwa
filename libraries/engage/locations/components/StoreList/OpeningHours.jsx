import React from 'react';
import PropTypes from 'prop-types';
import { i18n } from '@shopgate/engage/core';
import { openingHours, openingHoursRow, openingHoursDay } from './style';

/**
 * Renders the store's opening hours.
 * @param {Object} props The compnent props.
 * @returns {JSX}
 */
function OpeningHours({ hours }) {
  if (!hours) {
    return null;
  }

  return (
    <div className={openingHours}>
      {Object.keys(hours).map((key) => {
        if (!hours[key] || hours[key] === '') {
          return null;
        }

        return (
          <div className={openingHoursRow} key={key}>
            <div className={openingHoursDay}>
              {i18n.text(`product.location.${key}`)}
            </div>
            <div data-test-id={`hours-${key}`}>{hours[key]}</div>
          </div>
        );
      })}
    </div>
  );
}

OpeningHours.propTypes = {
  hours: PropTypes.shape(),
};

OpeningHours.defaultProps = {
  hours: null,
};

export default OpeningHours;
