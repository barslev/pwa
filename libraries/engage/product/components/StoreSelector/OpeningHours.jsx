import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import { openingHours, openingHoursRow, openingHoursDay } from './style';

/**
 * Renders the store#s opening hours in the store details.
 * @param {Object} props The compnent props.
 * @returns {JSX}
 */
function OpeningHours({ hours }) {
  if (!hours) {
    return null;
  }

  return (
    <div className={openingHours}>
      {Object.keys(hours).map((key) => {
        if (!hours[key] || hours[key] === '') {
          return null;
        }

        return (
          <div className={openingHoursRow} key={key}>
            <div className={openingHoursDay}>
              <I18n.Text string={`product.location.${key}`} />
            </div>
            <div data-test-id={`hours-${key}`}>{hours[key]}</div>
          </div>
        );
      })}
    </div>
  );
}

OpeningHours.propTypes = {
  hours: PropTypes.shape(),
};

OpeningHours.defaultProps = {
  hours: null,
};

export default OpeningHours;
