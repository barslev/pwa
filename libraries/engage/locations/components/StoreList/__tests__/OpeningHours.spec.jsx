import React from 'react';
import { shallow } from 'enzyme';
import OpeningHours from '../OpeningHours';

const hours = {
  mon: '8:00am - 4:00pm',
  tue: '',
  wen: '8:00am - 8:00pm',
};

describe('<OpeningHours />', () => {
  it('should not render if no hours are passed', () => {
    const wrapper = shallow(<OpeningHours />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should not render hours that are empty', () => {
    const wrapper = shallow(<OpeningHours hours={hours} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('[data-test-id="hours-mon"]').text()).toEqual(hours.mon);
    expect(wrapper.find('[data-test-id="hours-tue"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="hours-wen"]').text()).toEqual(hours.wen);
  });
});
