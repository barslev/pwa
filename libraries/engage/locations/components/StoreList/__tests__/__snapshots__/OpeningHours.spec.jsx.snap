// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpeningHours /> should not render hours that are empty 1`] = `
<div
  className={
    Object {
      "data-css-1xm8nj0": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-9shxm7": "",
      }
    }
    key="mon"
  >
    <div
      className={
        Object {
          "data-css-1ppc7ry": "",
        }
      }
    >
      product.location.mon
    </div>
    <div
      data-test-id="hours-mon"
    >
      8:00am - 4:00pm
    </div>
  </div>
  <div
    className={
      Object {
        "data-css-9shxm7": "",
      }
    }
    key="wen"
  >
    <div
      className={
        Object {
          "data-css-1ppc7ry": "",
        }
      }
    >
      product.location.wen
    </div>
    <div
      data-test-id="hours-wen"
    >
      8:00am - 8:00pm
    </div>
  </div>
</div>
`;

exports[`<OpeningHours /> should not render if no hours are passed 1`] = `""`;
